app/scripts/lib/rpc-method-middleware/handlers/request-accounts.js

```diff
@@ -85,10 +88,15 @@ async function requestEthereumAccountsHandler(
 
   // If no accounts, request the accounts permission
   try {
+    locks.add(origin);
+    await getUnlockPromise(true);
     await requestAccountsPermission();
   } catch (err) {
     res.error = err;
+    locks.delete(origin);
```

Comment:
This seems to be redundant - the `locks.delete(origin)` in the `finally` block will run as well